### DISCUSSION TITLE
ActionScript  Sftp Library

### DISCUSSION THREAD

**Discussion by santhoshpalan** [airsdk/Adobe-Runtime-Support#1789] [Q&A]
Sorry for posting this question AIR community.

Could anyone help me with the below?

1. Does Adobe still support Action Script language? Do they continue to release security fixes, new features/functionalities for it? Action Script 3.0 was released in 2006. There was no AS 4.0 release in the recent past. Does this mean Adobe has stopped supporting it (End of Life)?
2. I have a desktop application written in AcriptScript (packaged as AIR), which connects to the FTP server. However, now the target server supports only sftp. Is there any ActionScript sftp library ( I could not find any !)?

**Comment by ajwfrost** [comment]
Hi

The ActionScript language itself isn't something that gets security fixes as it's just a language ... we have started to update this again with some improvements and new language features but it won't really be called "ActionScript 4". Adobe stopped supporting it directly, but we picked it up (see https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air)

If you've got an app that's connecting to an FTP server then I'm guessing you're using a library to do the communications because AIR doesn't natively support the FTP protocol .. so it would then be a case of whether you're able to find a library for SFTP - it should be possible I think because we do have support for secure sockets, but I'm not aware of any... 

thanks

**Comment by ylazy** [comment]
You can use AS3 NativeProcess + Cmd + WinSCP, that is support SFTP connection.
